### DISCUSSION TITLE
Inherit docstring from nearest parent class

### DISCUSSION THREAD

**Discussion by HugoMVale** [mkdocstrings/python#117] [Q&A]
I often encounter the situation where I have an abstract class, where I specify abstract methods and properties, which are then implemented in child classes. For example something like:

```py
from abc import ABC, abstractmethod


class Parent(ABC):

    @abstractmethod
    def foo(self, x: int) -> int:
        """Take a number and return another.

        Parameters
        ----------
        x : int
            Input number.

        Returns
        -------
        int
            Output number.
        """
        pass


class Double(Parent):

    def foo(self, x: int) -> int:
        return 2*x


class Triple(Parent):

    def foo(self, x: int) -> int:
        return 3*x
```

If I hover the cursor over the child `foo()` methods - which are not documented -, VS code will automatically inherit the documentation from `Parent.foo()`. I very much like this feature, as it avoids repeating documentation for the "same" method.
 
Is it possible to apply the same principle with mkdocstrings, i.e. inherit the docstring from the nearest parent class?

**Comment by pawamoy** [comment]
Hello, now that we reached our [$500/month funding goal](https://pawamoy.github.io/insiders/#exec-5--500-plasmavac-user-guide), the [griffe-inherited-docstrings](https://mkdocstrings.github.io/griffe-inherited-docstrings/) extension is available to everyone, and I believe it does what you want :slightly_smiling_face:
